tests/main/test-user-daemon-in-graphical-environment/task.yaml

```
@@ -0,0 +1,42 @@
+summary: Ensure that the graphical user daemons work as expected
+
+details: |
+    This test ensures that an user daemon with the desktop, x11 or wayland plugs
+    will be launched with the destkop session only.
+
+# Only classic Ubuntu has the pulseaudio mediation patches. Ubuntu 14.04 is unsupported on the desktop.
+# TODO: extend tests.session and run this test on 14.04 as well.
+systems: [ubuntu-16.04-*, ubuntu-18.04-*, ubuntu-2*]
+
+prepare: |
+    "$TESTSTOOLS"/snaps-state install-local ./test-snapd-implicit-graphical-user-daemon
```

This didn't match the snap name before so prepare failed when testing on my machine. But let's move the test snap to `tests/lib/snaps/` and then this should work:

```suggestion
    "$TESTSTOOLS"/snaps-state install-local test-snapd-implicit-graphical-user-daemon
```